commander/src/bootstrapping/commands/generator/enable.ts

```diff
@@ -13,43 +13,94 @@
+const isLessThanZero = (value: number | undefined | null): boolean =>
+	value === null || value === undefined || value < 0;
+
+interface Status {
+	readonly address: string;
+	readonly height: number;
+	readonly maxHeightGenerated: number;
+	readonly maxHeightPrevoted: number;
+}
+
+export abstract class EnableCommand extends BaseGeneratorCommand {
+	static description = 'Enable block generation for given validator address.';
 
 	static examples = [
-		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu 100 100 10',
-		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu 100 100 10 --overwrite',
-		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu 100 100 10 --data-path ./data',
-		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu 100 100 10 --data-path ./data --password your_password',
+		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu --use-status-value',
+		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu --height=100 --max-height-generated=30 --max-height-prevoted=10',
+		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu --height=100 --max-height-generated=30 --max-height-prevoted=10 --data-path ./data',
+		'generator:enable lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu --height=100 --max-height-generated=30 --max-height-prevoted=10 --data-path ./data --password your_password',
 	];
 
 	static flags = {
-		...BaseForgingCommand.flags,
+		...BaseGeneratorCommand.flags,
+		height: Flags.integer({
+			description: 'Last generated block height.',
+		}),
+		'max-height-generated': Flags.integer({
+			description: 'Validators largest previously generated height.',
+		}),
+		'max-height-prevoted': Flags.integer({
+			description: 'Validators largest prevoted height for a block.',
+		}),
+		'use-status-value': Flags.boolean({
+			description: 'Use status value from the connected node',
+		}),
 	};
 
-	static args = [
-		...BaseForgingCommand.args,
-		{
-			name: 'height',
-			required: true,
-			description: 'Last forged block height.',
-		},
-		{
-			name: 'maxHeightGenerated',
-			required: true,
-			description: 'Validators largest previously forged height.',
-		},
-		{
-			name: 'maxHeightPrevoted',
-			required: true,
-			description: 'Validators largest prevoted height for a block.',
-		},
-	];
+	static args = [...BaseGeneratorCommand.args];
+
+	async run(): Promise<void> {
+		const { args, flags } = await this.parse(EnableCommand);
+		const { address } = args as { address: string };
+		const { 'use-status-value': useStatusValue } = flags;
+
+		let {
+			height,
+			'max-height-generated': maxHeightGenerated,
+			'max-height-prevoted': maxHeightPrevoted,
+		} = flags;
+
+		if (
+			!useStatusValue &&
+			(isLessThanZero(height) ||
+				isLessThanZero(maxHeightGenerated) ||
+				isLessThanZero(maxHeightPrevoted))
+		) {
+			throw new Error(
+				'The maxHeightGenerated and maxHeightPrevoted parameter value must be greater than or equal to 0',
```

Comment:
Since flags are provided in kebab-case format, wouldn't it be more consistent [and less confusing for a user] to use the same format in error message and test case names, rather than camelCase:

```
'The max-height-generated and max-height-prevoted parameter value must be greater than or equal to 0'
```

Also, it would be more specific to refer to them as _flag values_, rather than _parameter values_.